pkg/client/retrieve.go

```diff
@@ -40,13 +40,22 @@ var tarCommand = []string{
 	fmt.Sprintf("tar cf - %s/*.tar.gz", config.MasterResultsPath),
 }
 
-func (c *SonobuoyClient) RetrieveResults(cfg *RetrieveConfig) (io.Reader, <-chan error) {
+func (c *SonobuoyClient) RetrieveResults(cfg *RetrieveConfig) (io.Reader, <-chan error, error) {
+	if cfg == nil {
+		return nil, nil, errors.New("nil RetrieveConfig provided")
+	}
+
+	if err := cfg.Validate(); err != nil {
+		return nil, nil, errors.Wrap(err, "config validation failed")
+	}
+
 	ec := make(chan error, 1)
 	client, err := c.Client()
 	if err != nil {
 		ec <- err
-		return nil, ec
+		return nil, nil, err
```

Comment:
In this case there isn't a need to put the err on the channel (line above this) since we aren't even returning it.

Reply:
Good spot. I missed that!